modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3366MultipleGatewayPublishTestCase.java

```diff
@@ -0,0 +1,206 @@
+	public void testPublishApiWithDeSelectEnvironment() throws Exception {
+		driver.findElement(By.id("username")).clear();
+		driver.findElement(By.id("username")).sendKeys(userInfo.getUserName());
+		driver.findElement(By.id("pass")).clear();
+		driver.findElement(By.id("pass")).sendKeys(userInfo.getPassword());
+		driver.findElement(By.id("loginButton")).click();
+		wait.until(ExpectedConditions.visibilityOfElementLocated(By.linkText("Add")));
+		driver.findElement(By.linkText("Add")).click();
+		driver.findElement(By.id("name")).clear();
+		driver.findElement(By.id("name")).sendKeys("publishWithEnvironments3");
+		driver.findElement(By.id("context")).clear();
+		driver.findElement(By.id("context")).sendKeys("publishWithEnvironments3");
+		driver.findElement(By.id("version")).clear();
+		driver.findElement(By.id("version")).sendKeys("1.0.0");
+		driver.findElement(By.id("description")).clear();
+		driver.findElement(By.id("description")).sendKeys(API_DESCRIPTION);
+		driver.findElement(By.id("resource_url_pattern")).clear();
+		driver.findElement(By.id("resource_url_pattern")).sendKeys("*");
+		driver.findElement(By.id("inputResource")).clear();
+		driver.findElement(By.id("inputResource")).sendKeys("default");
+		driver.findElement(By.cssSelector("input.http_verb_select")).click();
+		driver.findElement(By.id("add_resource")).click();
+		driver.findElement(By.id("go_to_implement")).click();
+		wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("go_to_manage")));
+		driver.findElement(By.id("jsonform-0-elt-production_endpoints")).clear();
+		driver.findElement(By.id("jsonform-0-elt-production_endpoints")).sendKeys(API_URL);
+		driver.findElement(By.id("go_to_manage")).click();
+		wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("publish_api")));
+		driver.findElement(By.xpath("//button[@type='button']")).click();
+		driver.findElement(By.xpath("//input[@value='Gold']")).click();
+		driver.findElement(By.cssSelector("legend.legend-with-hidden-info.js_hidden_section_title")).click();
+		WebElement checkBox = driver.findElement(By.cssSelector("input.env"));
+		if (checkBox.isSelected()) {
+			checkBox.click();
+		}
+		driver.findElement(By.id("publish_api")).click();
+		Thread.sleep(2000);
+		Assert.assertTrue(!isAPIPublished("publishWithEnvironments3", "1.0.0"),
+		                  "API is Successfully published with de select environment in environment section");
+		driver.findElement(By.id("userMenu")).click();
+		driver.findElement(By.cssSelector("button.btn.btn-danger")).click();
+	}
+
+	@AfterClass(alwaysRun = true)
+	public void tearDown() throws Exception {
+		driver.quit();
+	}
+
+	private boolean isAPIPublished(String apiName, String version) throws IOException {
+		BufferedReader input = new BufferedReader(new FileReader(carbonLogFilePath));
+		String lastLine = null, line;
+
+		while ((line = input.readLine()) != null) {
+			lastLine = line;
+		}
+		input.close();
```

Comment:
Since this is a test case, the impact might be low, however as a best practice, close all such streams within a finally block to avoid possible memory leaks.